haskellPackages.blank-canvas: 0.6.3 -> 0.6.2

endgame opened the pull request:
###### Motivation for this change

blank-canvas-0.6.3 depends on base-compat-batteries-0.10, which
depends on base-compat-0.10. This conflicts with the rest of the LTS
set, which uses base-compat-0.9. No base-compat-batteries-0.9 exists.

blank-canvas-0.6.2 only depends on base-compat >= 0.8 && < 0.10.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

endgame committed:
haskellPackages.blank-canvas: 0.6.3 -> 0.6.2

blank-canvas-0.6.3 depends on base-compat-batteries-0.10, which
depends on base-compat-0.10. This conflicts with the rest of the LTS
set, which uses base-compat-0.9. No base-compat-batteries-0.9 exists.

blank-canvas-0.6.2 only depends on base-compat >= 0.8 && < 0.10.

peti commented:
Wouldn't it make more sense to add `base-compat-batteries == 0.10.*` so that we can build the latest version of `blank-canvas` instead?

endgame commented:
Quite possible. I held back `base-compat-batteries` because `base-compat-0.9.3` is still what's being used in stackage nightlies.

I made a test stack project on LTS-11.7, added `base-compat-0.10.1` to `extra-deps` and added every revdep stackage knew about that was in the LTS snapshot and tried to build:

```
In the dependencies for aeson-1.2.4.0:
    base-compat-0.10.1 from stack configuration does not match >=0.9.1 && <0.10 (latest matching version is 0.9.3)
needed due to z-0.1.0.0 -> aeson-1.2.4.0

In the dependencies for haxr-3000.11.2:
    base-compat-0.10.1 from stack configuration does not match >=0.8 && <0.10 (latest matching version is 0.9.3)
needed due to z-0.1.0.0 -> haxr-3000.11.2

In the dependencies for rest-core-0.39:
    base-compat-0.10.1 from stack configuration does not match >=0.8 && <0.10 (latest matching version is 0.9.3)
needed due to z-0.1.0.0 -> rest-core-0.39

In the dependencies for rest-snap-0.3.0.0:
    base-compat-0.10.1 from stack configuration does not match >=0.8 && <0.10 (latest matching version is 0.9.3)
needed due to z-0.1.0.0 -> rest-snap-0.3.0.0

In the dependencies for rest-types-1.14.1.2:
    base-compat-0.10.1 from stack configuration does not match >=0.8 && <0.10 (latest matching version is 0.9.3)
needed due to z-0.1.0.0 -> rest-types-1.14.1.2

In the dependencies for tdigest-0.1:
    base-compat-0.10.1 from stack configuration does not match >=0.9.1 && <0.10 (latest matching version is 0.9.3)
needed due to z-0.1.0.0 -> tdigest-0.1

In the dependencies for wl-pprint-text-1.1.1.0:
    base-compat-0.10.1 from stack configuration does not match >=0.9 && <0.10 (latest matching version is 0.9.3)
needed due to z-0.1.0.0 -> wl-pprint-text-1.1.1.0
```

So it seems like a decent number of packages haven't fixed their bounds for `base-compat-0.10`. Which way do you want me to take this? Holding back `blank-canvas` or pushing a bunch of other stuff forward?

peti commented:
I see! Well, I think you should choose whatever solution works best for you. Maybe it's too soon to upgrade to `base-compat-batteries == 0.10.x`.
